src/features/pass-survey/model/tests/AlertsExtractor.extractFromItem.test.ts

```diff
@@ -0,0 +1,134 @@
+import { ILogger } from '@app/shared/lib';
+
+import { Answer, PipelineItem } from '../../lib';
+import { AlertsExtractor } from '../AlertsExtractor';
+
+jest.mock('@app/shared/lib/constants', () => ({
+  ...jest.requireActual('@app/shared/lib/constants'),
+  STORE_ENCRYPTION_KEY: '12345',
+}));
+
+const mockExtractFunctions = (extractor: AlertsExtractor) => {
+  const extractFromRadioMock = jest.fn();
+  const extractFromCheckboxMock = jest.fn();
+  const extractFromStackedRadioMock = jest.fn();
+  const extractFromStackedCheckboxMock = jest.fn();
+  const extractFromSliderMock = jest.fn();
+  const extractFromStackedSliderMock = jest.fn();
+
+  extractor.extractFromCheckbox = extractFromCheckboxMock;
+  extractor.extractFromRadio = extractFromRadioMock;
+  extractor.extractFromSlider = extractFromSliderMock;
+  extractor.extractFromStackedCheckbox = extractFromStackedCheckboxMock;
+  extractor.extractFromStackedRadio = extractFromStackedRadioMock;
+  extractor.extractFromStackedSlider = extractFromStackedSliderMock;
+
+  return {
+    extractFromRadioMock,
+    extractFromCheckboxMock,
+    extractFromStackedRadioMock,
+    extractFromStackedCheckboxMock,
+    extractFromSliderMock,
+    extractFromStackedSliderMock,
+  };
+};
```

Comment:
I have a couple of questions here:

Why do we need to return the links to the same methods of the extractor object? In my opinion, there is no need to do that at all. We can assign mocking functions to the extractor and then call these functions using their original name (e.g. extractFromCheckbox) in the "it" tests

Can we move the function to the `beforeAll` function and use it instead of `beforeEach`? I don't see a reason why the extractor should be recreated for each tests

Example:
```
...
describe('AlertsExtractor: test extractFromItem', () => {
  let extractor: AlertsExtractor;

  beforeAll(() => {
    extractor = new AlertsExtractor({
      log: jest.fn(),
      warn: jest.fn(),
      error: jest.fn(),
      info: jest.fn(),
    } as unknown as ILogger);

    const extractFromRadioMock = jest.fn();
    const extractFromCheckboxMock = jest.fn();
    const extractFromStackedRadioMock = jest.fn();
    const extractFromStackedCheckboxMock = jest.fn();
    const extractFromSliderMock = jest.fn();
    const extractFromStackedSliderMock = jest.fn();

    extractor.extractFromCheckbox = extractFromCheckboxMock;
    extractor.extractFromRadio = extractFromRadioMock;
    extractor.extractFromSlider = extractFromSliderMock;
    extractor.extractFromStackedCheckbox = extractFromStackedCheckboxMock;
    extractor.extractFromStackedRadio = extractFromStackedRadioMock;
    extractor.extractFromStackedSlider = extractFromStackedSliderMock;
  });

  it('Should call extractFromRadio when type is Radio', () => {
    //@ts-expect-error
    extractor.extractFromItem({ type: 'Radio' } as PipelineItem, {} as Answer);

    expect(extractor.extractFromRadio).toBeCalledTimes(1);
  });
...
```

Reply:
Maybe, there is no need to even use `beforeAll` at all because the extractor variable is already defined in the scope of the `describe` function

Reply:
1. ok, thanks, maube "return" doesn't make sence.
2. Just a good practice for achieving isolation between tests. Theoretically, a tested instance could keep and update any state during tests that'd fail other tests. To analyze and fix such errors would be difficult enough. The goal here is to test an instance's single function in isolation. For the whole component testing, we implement tests that named "penetrating".